### PR TITLE
Create path in log bundle for unstarted containers

### LINKED ISSUE #2857: vicadmin: not collecting logs for unstarted containers
vicadmin is not collecting logs for containers if output.log doesn't exist.
I assume the same is likely true for tether.debug.

This is problematic if the file you were after was vmware.log for a container that didn't get far enough for output. The resulting log bundle did not contain a directory for the container.

```
time=2016-10-22T03:05:12.120714213Z level=debug msg=[ END ] [main.listVMPaths:276] [36.680786ms]  
time="2016-10-22T03:05:12Z" level=info msg="Processed File read Path : ghicken-test_2/output.log" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File write Path : ghicken-test/output.log" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File read Path : ghicken-test_2/vmware.log" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File write Path : ghicken-test/vmware.log" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File read Path : ghicken-test_2/tether.debug" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File write Path : ghicken-test/tether.debug" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File read Path : elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File write Path : elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File read Path : elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/vmware.log" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File write Path : elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/vmware.log" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File read Path : elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/tether.debug" 
time="2016-10-22T03:05:12Z" level=info msg="Processed File write Path : elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/tether.debug" 
time=2016-10-22T03:05:12.136374856Z level=debug msg=[ END ] [main.findDatastoreLogs:326] [52.456518ms]  
time=2016-10-22T03:05:12.136621458Z level=debug msg=[BEGIN] [main.findDiagnosticLogs:171] 
time=2016-10-22T03:05:12.136721383Z level=debug msg=[ END ] [main.findDiagnosticLogs:171] [99.447Âµs]  
time=2016-10-22T03:05:12.136804535Z level=debug msg=[BEGIN] [main.(*server).bundleLogs:219] 
time=2016-10-22T03:05:12.136972118Z level=debug msg=[BEGIN] [main.zipEntries:266] 
time="2016-10-22T03:05:12Z" level=info msg="Collecting log with reader elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log(main.datastoreReader{ds:(*object.Datastore)(0xc4204e1a40), path:\"elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log\"})" 
time=2016-10-22T03:05:12.137170275Z level=debug msg=[BEGIN] [main.datastoreReader.open:367] elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log 
time=2016-10-22T03:05:12.204509996Z level=debug msg=[BEGIN] [main.httpEntry:231] elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log 
time=2016-10-22T03:05:12.204896704Z level=debug msg=[ END ] [main.httpEntry:231] [387.994Âµs] elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log 
time=2016-10-22T03:05:12.205045987Z level=debug msg=[ END ] [main.datastoreReader.open:367] [67.876475ms] elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log 
time="2016-10-22T03:05:12Z" level=warning msg="error reading elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log({%!s(*object.Datastore=&{{/ha-datacenter/datastore/datastore1 0xc420219680 {Datastore 56a2eeef-3cafae6e-2bd3-000c29b00df4}}}) elated_mclean-c73148967c7e0f5822697f5aeb971b5f1692a7450811cf4afd245e51dfcbfb7e/output.log}): 404 Not Found\n" 
time="2016-10-22T03:05:12Z" level=info msg="Collecting log with reader df(\"df\")" 
time=2016-10-22T03:05:12.205420370Z level=debug msg=[BEGIN] [main.commandReader.open:178] df 
time=2016-10-22T03:05:12.206969680Z level=debug msg=[ END ] [main.commandReader.open:178] [1.539924ms] df 
time="2016-10-22T03:05:12Z" level=info msg="df has size 381" 
```

### PR DESCRIPTION
**Pull request by sgairo**
Fixes #2857

### CODE REVIEW

File: cmd/vicadmin/logs.go [hunk 1]
```diff
@@ -230,114 +230,123 @@ func findDiagnosticLogs(c *session.Session) (map[string]entryReader, error) {
...
+        if err != nil {
+            log.Errorf("Failed to write header for %s: %s", header.Name, err)
+            continue
+        }
+
+        log.Infof("%s has size %d", header.Name, header.Size)
+
+        // be explicit about the number of bytes to copy as the log files will likely
+        // be written to during this exercise
+        if e != nil{
```
Comment (anchal-agrawal) [inline cmd/vicadmin/logs.go]: gofmt: 

```go
if e != nil {
```

File: cmd/vicadmin/logs.go [hunk 2]
```diff
@@ -230,114 +230,123 @@ func findDiagnosticLogs(c *session.Session) (map[string]entryReader, error) {
...
+
+    for name, r := range readers {
+        log.Infof("Collecting log with reader %s(%#v)", name, r)
+
+        e, err := r.open()
+        if err != nil {
+            log.Warningf("error reading %s(%s): %s\n", name, r, err)
+        }
+        var sz int64
+        if e != nil{
```
Comment (anchal-agrawal) [inline cmd/vicadmin/logs.go]: ditto

File: cmd/vicadmin/logs.go [hunk 3]
```diff
@@ -230,114 +230,123 @@ func findDiagnosticLogs(c *session.Session) (map[string]entryReader, error) {
...
+            continue
+        }
+    }
+
+    _ = t.Flush()
+    _ = w.Close()
+    wg.Wait()
+    _ = r.Close()
+
+    return nil
```
Comment (gigawhitlocks) [inline cmd/vicadmin/logs.go]: Why is the indent width changed? @sgairo can you run `gofmt` on these files please?

Comment (sgairo) [inline cmd/vicadmin/logs.go]: Fixed it :+1:

File: cmd/vicadmin/logs.go [hunk 4]
```diff
@@ -255,10 +255,14 @@ func tarEntries(readers map[string]entryReader, out io.Writer) error {
 			log.Warningf("error reading %s(%s): %s\n", name, r, err)
 			continue
 		}
+		var sz int64
+		if e != nil {
+			sz = e.Size()
```
Comment (gigawhitlocks) [inline cmd/vicadmin/logs.go]: Is there a reason I'm not seeing not to just do `sz := e.Size()` ?

File: cmd/vicadmin/logs.go [hunk 5]
```diff
@@ -255,10 +255,14 @@ func tarEntries(readers map[string]entryReader, out io.Writer) error {
 			log.Warningf("error reading %s(%s): %s\n", name, r, err)
 			continue
 		}
+		var sz int64
```
Comment (gigawhitlocks) [inline cmd/vicadmin/logs.go]: What is the behavior when `e` is `nil`? `sz` gets set to `nil` below also? Is that going to cause a nil pointer exception somewhere else?

Comment (sgairo) [inline cmd/vicadmin/logs.go]: if e is nil, sz will be 0. If e is nil then e.Size() won't work so we work with 0 instead.

Comment (gigawhitlocks) [inline cmd/vicadmin/logs.go]: ah right unassigned int is 0

**Comment by gigawhitlocks** [comment]
Can we get an integration or regression test? Should be good to go after that

**Pull request merged**